test_main.py

```diff
@@ -216,3 +216,76 @@ def test_register_fail(name_i: int, room_i: int, phone_i: int, email_i: int, use
                 headers=generate_auth_headers(username=resident_usernames[username_i], password=resident_passwords[password_i]).model_dump(),
             )
             assert response.status_code == status.HTTP_404_NOT_FOUND
+
+
+@pytest.mark.parametrize("pass_i", range(13))
+@pytest.mark.parametrize("fail_i", range(13))
+def test_register_username_taken(pass_i: int, fail_i: int) -> None:
+    taken_username = []
+    for iterations in range(1, pass_i + 2):
+        name = f"test-{random_string(random.randint(1, 69))}"
+        room = random.randint(0, 32767)
+        now = datetime.now(timezone.utc)
+        birthday = datetime(now.year - 18, now.month, now.day, tzinfo=timezone.utc)
+        phone = random_numstring(random.randint(1, 15))
+        email = f"{random_string(random.randint(1, 69))}@{random_string(random.randint(1, 69))}.com"
+        username = random_string(random.randint(1, 255))
+        password = random_string(random.randint(8, 255))
+        taken_username.append(username)
+        with TestClient(app) as client:
+            response = client.post(
+                "/api/v1/register",
+                params={
+                    "name": name,
+                    "room": room,
+                    "birthday": birthday.isoformat(),
+                    "phone": phone,
+                    "email": email,
+                },
+                headers=generate_auth_headers(username=username, password=password).model_dump(),
+            )
+            assert response.status_code == status.HTTP_200_OK
+    for iterations in range(1, fail_i + 2):
```

Comment:
Add an empty line between 2 code blocks.
```suggestion

    for iterations in range(1, fail_i + 2):
```